macros/new_datacard_tt_ot.py

```diff
@@ -190,6 +207,287 @@
+                cutstring = signal_region_MC_nofakes_DY
+            else:
+                datasets = cfg.datasets.DY_datasets
+                cutstring = cfg.sys_dict[sys]['change']
+            ZL_cfgs = build_cfgs(
+                [dataset.name+'_ZL' for dataset in datasets], 
+                datasets, variable,
+                cutstring & cfg.cuts_datacards['ZL'], bins)
+            ZL_comp = merge_cfgs('ZL', ZL_cfgs)
+            dc_comps[variable][sys]['ZL'] = ZL_comp
+
+        if 'ZJ' in cfg.sys_dict[sys]['processes']:
+            if cfg.sys_dict[sys]['change'] == 'dataset':
+                datasets = cfg.sys_datasets[sys]
+                cutstring = signal_region_MC_nofakes_DY
+            else:
+                datasets = cfg.datasets.DY_datasets
+                cutstring = cfg.sys_dict[sys]['change']
+            ZJ_cfgs = build_cfgs(
+                [dataset.name+'_ZJ' for dataset in datasets], 
+                datasets, variable,
+                cutstring & cfg.cuts_datacards['ZJ'], bins)
+            ZJ_comp = merge_cfgs('ZJ', ZJ_cfgs)
+            dc_comps[variable][sys]['ZJ'] = ZJ_comp
+
+        if 'ZLL' in cfg.sys_dict[sys]['processes']:
+            ZLL_comp = merge_comps('ZLL', [ZL_comp, ZJ_comp])
+            dc_comps[variable][sys]['ZLL'] = ZLL_comp
+
+        if 'DY' in cfg.sys_dict[sys]['processes']:
+            if cfg.sys_dict[sys]['change'] == 'dataset':
+                datasets = cfg.sys_datasets[sys]
+                cutstring_l1 = signal_region_MC_nofakes_DY
+                cutstring_l2 = signal_region_MC_nofakes_DY
+            else:
+                datasets = cfg.datasets.DY_datasets
+                cutstring_l1 = cfg.sys_dict[sys]['fakecutstring_l1']
+                cutstring_l2 = cfg.sys_dict[sys]['fakecutstring_l2']
+            fake_cfgs_DY_1 = build_cfgs(['fakesDY1'], datasets, variable, cutstring_l1, bins)
+            fake_cfgs_DY_2 = build_cfgs(['fakesDY2'], datasets, variable, cutstring_l2, bins)
+            fake_cfgs_DY_1['scale'] = -1.
+            fake_cfgs_DY_2['scale'] = -1.
+            sys_fake_comp['DY'] = merge_cfgs('DY', [fake_cfgs_DY_1,fake_cfgs_DY_2])
+            
+        if 'TTT' in cfg.sys_dict[sys]['processes']:  
+            if cfg.sys_dict[sys]['change'] == 'dataset':
+                datasets = cfg.sys_datasets[sys]
+                cutstring = signal_region_MC_nofakes_TT
+            else:
+                datasets = cfg.datasets.TT_datasets
+                cutstring = cfg.sys_dict[sys]['change']  
+            TTT_cfgs = build_cfgs(
+                [dataset.name+'_TTT' for dataset in datasets], 
+                datasets, variable,
+                cutstring & cfg.cuts_datacards['TTT'], bins)
+            TTT_comp = merge_cfgs('TTT', TTT_cfgs)
+            dc_comps[variable][sys]['TTT'] = TTT_comp
+
+        if 'TTJ' in cfg.sys_dict[sys]['processes']:    
+            if cfg.sys_dict[sys]['change'] == 'dataset':
+                datasets = cfg.sys_datasets[sys]
```

Comment:
Are we sure this is safe to use a one-depth dictionary to change datasets?
What if two cfgs have to change their datasets, but do not have the same?
For example, for a same systematic, assume DY and TT change datasets.
If this never happens then this code should be OK.